src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -466,21 +466,19 @@ class TypeComparer(initctx: Context) extends DotClass with ConstraintHandling {
       false
   }
 
-  /** If `projection` is of the form T # Apply where `T` is an instance of a Lambda class,
-   *  and `other` is not a type lambda projection, then convert `other` to a type lambda `U`, and
+  /** If `projection` is a hk projection T#$apply
+   *  and `other` is not a hk projection, then convert `other` to a hk projection `U`, and
    *  continue with `T <:< U` if `inOrder` is true and `U <:< T` otherwise.
    */
   def compareHK(projection: NamedType, other: Type, inOrder: Boolean) =
```

Comment:
Since you're changing `compareHK`, you may want to adapt and apply https://github.com/dotty-staging/dotty/commit/eac33f06cbc37327ded86dd51cae735f89f15519 too.